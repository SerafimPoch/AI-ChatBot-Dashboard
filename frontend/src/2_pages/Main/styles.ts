import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import { LAPTOP_MQ, TABLET_MQ } from "@/src/6_shared/media-queries";
import styled from "@emotion/styled";

export const MainContainer = styled.main`
  ${() => {
    const themeColors = useThemeContext();

    return `
      width: 100vw;
      padding: 20px 64px 20px 20px;
      background: ${themeColors?.coreBackground};
      display: flex;
      justify-content: center;
      gap: 64px;
      align-items: stretch;

      @media(max-width:${LAPTOP_MQ}px) {
        gap: 40px;
      }

      @media(max-width:${TABLET_MQ}px) {
        gap: 20px;
        padding: 20px;
      }
    `;
  }}
`;

export const Container = styled.div`
  width: 100%;
  max-width: 1440px;
`;

export const MainContentWrapper = styled.div`
  display: grid;
  grid-template-columns: minmax(700px, 1.5fr) minmax(500px, 1fr);
  grid-row-gap: 8px;
  grid-column-gap: 8px;

  @media (max-width: ${LAPTOP_MQ}px) {
    grid-template-columns: 1fr 1fr;
  }

  @media (max-width: ${TABLET_MQ}px) {
    grid-template-columns: 1fr;
  }
`;
