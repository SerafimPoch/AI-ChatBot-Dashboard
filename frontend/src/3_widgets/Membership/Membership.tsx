import { Bar, BarChart, ResponsiveContainer, XAxis, YAxis } from "recharts";
import {
  MembershipContainer,
  MembershipTitle,
  CustomBarComponent,
} from "./Membership.styles";
import { CHART_DATA } from "./constants";

export default function Membership() {
  return (
    <MembershipContainer>
      <MembershipTitle>Membership</MembershipTitle>
      <ResponsiveContainer width="100%" height={120}>
        <BarChart data={CHART_DATA} margin={{ left: -20 }}>
          <Bar
            dataKey="uv"
            shape={<CustomBarComponent fill="#FFCC6F" />}
            barSize={72}
          />
          <Bar
            dataKey="amt"
            shape={<CustomBarComponent fill="#EA7465" />}
            barSize={72}
          />
          <XAxis
            dataKey="date"
            tickLine={false}
            axisLine={false}
            tickMargin={8}
            tick={{ fill: "#A0A5A9" }}
          />
          <YAxis
            orientation="right"
            axisLine={false}
            tickLine={false}
            tick={{ fill: "#A0A5A9" }}
            tickFormatter={(value) => (value === 0 ? "" : value)}
          />
        </BarChart>
      </ResponsiveContainer>
    </MembershipContainer>
  );
}
