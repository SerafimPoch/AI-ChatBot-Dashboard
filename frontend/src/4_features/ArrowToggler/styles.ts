import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const ArrowTogglerContainer = styled.div`
  ${() => {
    const themeColors = useThemeContext();

    return `
      display: flex;
      flex-direction: column;
      align-items: center;
      justify-content: center;
      gap: 32px;
      align-self: stretch;
      border-radius: 12px;
      background: ${themeColors?.grey};
      width: 56px;
    `;
  }}
`;
