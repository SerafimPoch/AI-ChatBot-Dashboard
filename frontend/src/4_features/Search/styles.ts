import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import styled from "@emotion/styled";

export const SearchContainer = styled.div`
  position: relative;
  display: flex;
  align-items: center;
  gap: 8px;
`;

export const SearchInput = styled.input`
  ${() => {
    const themeColors = useThemeContext();

    return `
      background: transparent;
      border: 1px solid #5d74f1;
      outline: none;
      border-radius: 6px;
      padding: 0 5px;
      color: ${themeColors?.textPrimary};
    `;
  }}
`;
