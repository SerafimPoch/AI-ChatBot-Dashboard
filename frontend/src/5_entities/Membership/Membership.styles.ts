import { useThemeContext } from "@/src/6_shared/contexts/ThemeContext";
import { LAPTOP_MQ, MOBILE_MQ } from "@/src/6_shared/media-queries";
import styled from "@emotion/styled";

export const MembershipContainer = styled.div`
  border-radius: 32px;
  border: 2px solid #f2f2f2;
  grid-column: 2;
  padding: 24px 16px 24px 32px;

  @media (max-width: ${LAPTOP_MQ}px) {
    grid-row: 2;
    grid-column: 1;
  }

  @media (max-width: ${LAPTOP_MQ}px) {
    grid-column: 1/3;
  }

  @media (max-width: ${MOBILE_MQ}px) {
    padding: 24px 16px 24px 24px;
  }
`;

export const MembershipTitle = styled.p`
  ${() => {
    const themeColors = useThemeContext();

    return `
      font-family: var(--font-sf-pro-rounded);
      font-size: 20px;
      font-weight: 600;
      line-height: 24px;
      color: ${themeColors?.textPrimary};
      margin-bottom: 16px;

     
    `;
  }}
`;

export const CustomBarComponent = styled.rect`
  fill: ${({ color }) => color};
  rx: 12px;
  ry: 12px;
`;
