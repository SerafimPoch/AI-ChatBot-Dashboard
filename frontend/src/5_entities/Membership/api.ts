import { z } from "zod";
import redaxios from "redaxios";
import { useQuery } from "react-query";

const ChartItemSchema = z.object({
  id: z.number(),
  date: z.string(),
  uv: z.number(),
  amt: z.number(),
});

const UseChartSchema = z.array(ChartItemSchema);

export type IUseChart = z.infer<typeof UseChartSchema>;

export const useChart = () => {
  return useQuery<IUseChart>("chart", async () => {
    const { data } = await redaxios.get("http://localhost:3001/chart");

    if (!UseChartSchema.safeParse(data).success) {
      console.log("Invalid useChart api response structure");
      return [] as IUseChart;
    }

    return data;
  });
};
