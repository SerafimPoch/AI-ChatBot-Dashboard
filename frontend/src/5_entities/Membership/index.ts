export { default as Membership } from "./view";
