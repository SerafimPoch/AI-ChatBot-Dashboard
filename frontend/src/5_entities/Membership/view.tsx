import { Bar, BarChart, ResponsiveContainer, XAxis, YAxis } from "recharts";
import {
  MembershipContainer,
  MembershipTitle,
  CustomBarComponent,
} from "./styles";
import { useChart } from "./api";

export default function Membership() {
  const { data, isLoading } = useChart();

  return (
    <MembershipContainer>
      {isLoading ? (
        <MembershipTitle>Loading...</MembershipTitle>
      ) : (
        <>
          <MembershipTitle>Membership</MembershipTitle>
          <ResponsiveContainer width="100%" height={120}>
            <BarChart data={data} margin={{ left: -2 }}>
              <Bar
                dataKey="uv"
                shape={<CustomBarComponent fill="#FFCC6F" />}
                barSize={65}
              />
              <Bar
                dataKey="amt"
                shape={<CustomBarComponent fill="#EA7465" />}
                barSize={65}
              />
              <XAxis
                dataKey="date"
                tickLine={false}
                axisLine={false}
                tickMargin={8}
                tick={{ fill: "#A0A5A9" }}
              />
              <YAxis
                orientation="right"
                axisLine={false}
                tickLine={false}
                tick={{ fill: "#A0A5A9" }}
                tickFormatter={(value) => (value === 0 ? "" : value)}
              />
            </BarChart>
          </ResponsiveContainer>
        </>
      )}
    </MembershipContainer>
  );
}
