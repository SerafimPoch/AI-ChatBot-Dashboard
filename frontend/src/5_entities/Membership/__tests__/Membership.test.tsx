import React from "react";
import { render } from "@testing-library/react";
import { MembershipContainer, MembershipTitle } from "../Membership.styles";

describe("Membership page components", () => {
  it("MembershipContainer renders correctly", () => {
    const { asFragment } = render(<MembershipContainer />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("MembershipTitle renders correctly", () => {
    const { asFragment } = render(<MembershipTitle />);
    expect(asFragment()).toMatchSnapshot();
  });
});
