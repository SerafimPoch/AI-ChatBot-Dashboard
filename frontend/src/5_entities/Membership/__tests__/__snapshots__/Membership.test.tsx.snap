// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Membership page components MembershipContainer renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  border-radius: 32px;
  border: 2px solid #f2f2f2;
  grid-column: 2;
  padding: 24px 16px 24px 32px;
}

@media (max-width: 1550px) {
  .emotion-0 {
    grid-row: 2;
    grid-column: 1;
  }
}

@media (max-width: 1550px) {
  .emotion-0 {
    grid-column: 1/3;
  }
}

@media (max-width: 1000px) {
  .emotion-0 {
    padding: 24px 16px 24px 24px;
  }
}

<div
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Membership page components MembershipTitle renders correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-family: var(--font-sf-pro-rounded);
  font-size: 20px;
  font-weight: 600;
  line-height: 24px;
  color: undefined;
  margin-bottom: 16px;
}

<p
    class="emotion-0"
  />
</DocumentFragment>
`;
